### PR TITLE
Refactor authenticator, get /manifest listing.

### PR DESCRIPTION
**Pull request by mathewjordan**
This does a refactor on the App and Editor pages to tidy things up a bit, and then adds a GET request on the hardcoded API endpoint for `/manifests` route. The results of this is output in the table. To test, you're going to need a login in the AWS cognito user pool tied to the app. I'll send you the dummy user I created.

The goal of this work is just to get the basic convention for listing out results of this table.

Relates to first item in https://github.com/nulib/repodev_planning_and_docs/issues/4303

image

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -1,24 +1,42 @@
...
 import "../styles/app.css";
 
+import { Amplify } from "aws-amplify";
 import { AppProvider } from "../context/AppContext";
+import { Authenticator } from "@aws-amplify/ui-react";
 import Editor from "./Editor";
 import React from "react";
 import { Theme } from "@radix-ui/themes";
+import authenticatorComponents from "./Vendor/Amplify/Authenticator";
+import awsExports from "../aws-exports";
```
Comment (adamjarling) [inline src/components/App.tsx:11]: Assume the exported values in `awsExports` are safe to publicly expose?

File: src/components/Layouts/Collection.tsx [hunk 2]
```diff
@@ -1,30 +1,42 @@
...
+  const [manifests, setManifests] = useState<ManifestEditorManifest[]>([]);
 
-  const handleLogout: MouseEventHandler<HTMLButtonElement> = () => {
-    dispatch({ type: ActionTypes.LOGOUT });
-  };
+  const { state } = useAppContext();
+  const { authToken } = state;
+
+  useEffect(() => {
+    const baseUrl =
```
Comment (adamjarling) [inline src/components/Layouts/Collection.tsx:26]: Eventually `baseUrl` is defined into a `CONSTANT`, or `env` variable?

File: src/components/Layouts/Collection.tsx [hunk 3]
```diff
@@ -1,30 +1,42 @@
...
+      "https://67qdcv50f4.execute-api.us-east-1.amazonaws.com/prod";
+
+    fetch(`${baseUrl}/manifests`, {
```
Comment (adamjarling) [inline src/components/Layouts/Collection.tsx:29]: For eventual tests, you'll likely want to move this network request to an external function, ie. `fetchManifests()`, which can be mocked.

File: src/context/AppContext.tsx [hunk 4]
```diff
@@ -1,40 +1,30 @@
 import React, { ReactNode, createContext, useContext, useReducer } from "react";
 
-// Define the initial state for your context
+import { useAuthenticator } from "@aws-amplify/ui-react";
```
Comment (adamjarling) [inline src/context/AppContext.tsx:3]: These vendor provided auth hooks are a gift from the heavens.  🙏👍

**Review by adamjarling — APPROVED** [review]
Looks and works great.  A few comments, but they don't necessarily have to happen now.

File: src/context/AppContext.tsx (hunk 4 above)
Comment (mathewjordan) [reply src/context/AppContext.tsx:3]: Absolutely.

File: src/components/App.tsx (hunk 1 above)
Comment (mathewjordan) [reply src/components/App.tsx:11]: Yup!

**Pull request merged**